app/Http/Controllers/API/ReportApiController.php

```
@@ -470,30 +506,68 @@ public function getCashflowData(Request $request): JsonResponse
+            $isBaseCurrency = $summary->currency_id === $baseCurrency->id;
+            $rate = $this->getLatestRateFromMap(
+                $summary->currency_id,
+                Carbon::parse($summary->date),
+                $allRatesMap,
+                $baseCurrency->id
+            );
+
+            // Resolve the source month for the rate that was actually applied
+            $debugRateSourceMonth = null;
+            if (!$isBaseCurrency && $rate !== null && array_key_exists($summary->currency_id, $allRatesMap)) {
+                foreach ($allRatesMap[$summary->currency_id] as $rateDate => $rateValue) {
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$rateValue'. (undefined)

(UnusedLocalVariable)